components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/streaming/sse/SseResponseStreamInterceptor.java

```diff
@@ -102,13 +110,37 @@ private boolean handleThrottlingAndAnalytics(int eventCount, MessageContext axi2
             }
             throttlePublisherService.execute(
                     () -> SseUtils.publishNonThrottledEvent(eventCount, messageId, throttleInfo, propertiesMap));
+            if (APIUtil.isAnalyticsEnabled()) {
+                try {
+                    publishAnalyticsData(eventCount, axi2Ctx);
+                } catch (AnalyticsException e) {
+                    log.error("Error while publishing analytics data", e);
+                }
+            }
             return true;
         } else {
             log.error("Throttle object cannot be null.");
         }
         return true;
     }
 
+    private void publishAnalyticsData(int eventCount, MessageContext axi2Ctx) throws AnalyticsException {
+
+        Object responseEventProvider = axi2Ctx.getProperty(SSE_ANALYTICS_INFO);
+        if (responseEventProvider == null) {
+            log.error("SSE Analytics event provider is null.");
+            return;
+        }
+        SseResponseEventDataProvider provider = (SseResponseEventDataProvider) responseEventProvider;
+        provider.setResponseCode((int) axi2Ctx.getProperty(SynapseConstants.HTTP_SC));
+        GenericRequestDataCollector dataCollector = new GenericRequestDataCollector(provider);
+        int count = 1;
+        while (count <= eventCount) {
+            dataCollector.collectData();
+            count++;
+        }
```

Comment:
Can we use something like this?

for (int i = 0; i < eventCount; i++) {
    dataCollector.collectData();
}

Reply:
updated in https://github.com/wso2/carbon-apimgt/pull/9944/commits/48d18ec636cb74ef0dc94fcd92d7143b2aa0c639